tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Wcf/WcfCommon.cs

```diff
@@ -90,12 +91,13 @@ internal static Scope CreateScope<TRequestContext>(TRequestContext requestContex
                 var requestHeaders = requestMessage.Headers;
                 string action = requestHeaders.Action;
                 Uri requestHeadersTo = requestHeaders.To;
+                var cleanedUrl = requestHeadersTo != null ? UriHelpers.CleanUri(requestHeadersTo, removeScheme: false, tryRemoveIds: false) : null;
 
                 span.DecorateWebServerSpan(
                     resourceName: action ?? requestHeadersTo?.LocalPath,
```

Comment:
I think you can actually swap it so you clean the `resourceName` instead and (if we want to defer ASM questions) leave the `httpUrl` unchanged.

For more background, when we generate spans, it is the `resourceName` that determines the resources that users can drill down into in the UI. For that reason, `resourceName` is the field that we really want to normalize to reduce the amount of unique values we have. As much as possible, these would be replaced with routes or controllers/actions values. `http.url` on the other hand we only scrub slightly so that it maintains the host, port, path, etc. since the field is highly searchable and unique values may be really helpful data points for users.

Reply:
So my suggestion is to clean the `requestHeadersTo?.LocalPath`

Reply:
That makes sense to me! I'll swap it around to clean that `resourceName` instead of `httUrl`.
Thanks for that additional insight also on how the resource names are used and normalized!